### ISSUE TITLE
Consider using command-line git client for fetching instead of libgit2

### ISSUE THREAD

**Issue by rtimush** [coralogix/protofetch#40]
Similarly to what Cargo does when `net.git-fetch-with-cli` is enabled. 
This would make protofetch use the "system" authentication configuration, eliminating the need to provide protofetch-specific GIT_USERNAME/GIT_PASSWORD.

**Comment by jeanbza** [comment]
Yes, please do this. This was several years worth of issues over at jj-vcs/jj (ex https://github.com/jj-vcs/jj/pull/5228) and resulted in subprocess spawning approach which has dramatically increased its capability.
